imapserver/session.go

```diff
@@ -114,3 +114,18 @@ type SessionUnauthenticate interface {
 	// Authenticated state
 	Unauthenticate() error
 }
+
+// SessionAppendLimit is an IMAP session which has the same APPEND limit for
+// all mailboxes.
+type SessionAppendLimit interface {
+	Session
+
+	// AppendLimit returns the maximum size in bytes that can be uploaded to this server
+	// in an APPEND command.
+	AppendLimit() uint32
+
+	// DiscloseLimit indicates whether the limit should be advertised in the CAPABILITY
+	// response. If false, only "APPENDLIMIT" will be listed, without the actual limit.
+	// If true, "APPENDLIMIT=<limit>" will be listed.
+	DiscloseLimit() bool
```

Comment:
What's the purpose of this method? Servers can already reject literals which are too large in APPEND without advertising a limit (by returning an error).